main.py

```diff
@@ -160,94 +162,97 @@ def is_ignored(field, ignored_range):
                     return True
         return False
 
-    def consistency_ignored_check(self, flow):
-        """Check if the flow is in the list of flows ignored by consistency.
-
-        Check by `cookie` range and `table_id` range.
-        Return True if the flow is in the ignored range, otherwise return
-        False.
-        """
-        # Check by cookie
-        if self.is_ignored(flow.cookie, self.cookie_ignored_range):
-            return True
-
-        # Check by `table_id`
-        if self.is_ignored(flow.table_id, self.tab_id_ignored_range):
-            return True
-        return False
-
     @listen_to("kytos/of_core.flow_stats.received")
     def on_flow_stats_check_consistency(self, event):
         """Check the consistency of a switch upon receiving flow stats."""
-        if not ENABLE_CONSISTENCY_CHECK:
+        self.check_consistency(event.content["switch"])
+
+    def check_consistency(self, switch):
+        """Check consistency of stored and installed flows given a switch."""
+        if not ENABLE_CONSISTENCY_CHECK or not switch.is_enabled():
             return
-        switch = event.content["switch"]
-        if switch.is_enabled():
+        with self._storehouse_lock:
```

Comment:
agreed. the idea here was to try to get rid of the `deepcopy` and be able to read atomically, since this table can grow to thousands of entries in prod later on, but I haven't run benchmark tests, so we don't know with concrete data yet when this might become an issue. However, I didn't get rid of the `deepcopy` after inserting the lock so I didn't accomplish the original goal either. I'll revert this, it's probably also better to still have some concurrency, and as if we start seeing problems related to this copy (as we can afford to implement more stress tests) we can assess other solutions, wdyt? Thanks for catching this.

Reply:
I've reverted this on this commit 5dfa50ab3de3049a7a3afaa045e4f77560799bcb fyi @italovalcy

Reply:
Thanks @viniarck. LGTM.